### PR TITLE
redux migration

### PR DESCRIPTION
**Pull request by amiteshhh**

### CODE REVIEW

File: src/movie/Edit.js [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
 import { MOVIE_LIST, GENRES, EMPTY_MOVIE_FORM_DATA } from '../consts/MovieConst';
+import { saveMovie } from '../redux/actions';
+import { getMovieById } from '../redux/selectors';
 
 import './Edit.scss';
 
-export function EditMovie() {
+const mapDispatchToProps = { saveMovie };
+
+const _EditMovie = ({ saveMovie }) => {
```
Comment (alionaSipchuk) [inline src/movie/Edit.js:14]: pls remove underscore!!
it's soo bad pattern

File: src/movie/Edit.js [hunk 2]
```diff
@@ -237,4 +241,6 @@ export function EditMovie() {
       </form>
     </div>
   );
-}
+};
+
+export const EditMovie = connect(null, mapDispatchToProps)(_EditMovie);
```
Comment (alionaSipchuk) [inline src/movie/Edit.js:246]: let's do:

`export default connect(null, mapDispatchToProps)(EditMovie);`

File: src/redux/selectors.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      val1 = val1.join();
+      val2 = val2.join();
+    }
+    return val1.localeCompare(val2);
+  });
+};
+
+export const getMoviesByFilter = (state) => {
+  const { activeGenre, sortKey, query } = state;
+  let movies = [...MOVIE_LIST];
```
Comment (alionaSipchuk) [inline src/redux/selectors.js:31]: i think that initial movie data should be in the store! not in the selector, pls fix it)

File: src/movie/List.js [hunk 4]
```diff
@@ -136,7 +115,9 @@ export class MovieList extends Component {
   }
 }
 
-MovieList.propTypes = {
+_MovieList.propTypes = {
   query: PropTypes.string,
   onDetailsClick: PropTypes.func,
 };
+
+export const MovieList = connect(mapStateToProps, mapDispatchToProps)(_MovieList);
```
Comment (alionaSipchuk) [inline src/movie/List.js:123]: remove underscors everywhere!!!

File: src/movie/List.js [hunk 5]
```diff
@@ -111,18 +90,18 @@ export class MovieList extends Component {
...
             ))}
           </select>
         </div>
         <div className="container-side-margin text-white">
-          <strong>{this.state.filteredMovies.length}</strong> movies found.
+          <strong>{this.props.movies.length}</strong> movies found.
         </div>
         <div className="flex-grid movie-grid">
-          {this.state.filteredMovies.map((movie, i) => (
+          {this.props.movies.map((movie, i) => (
```
Comment (alionaSipchuk) [inline src/movie/List.js:104]: as i see you useed more that one time this.props.movies
pls get movies destructively from props, and use everywhere only movies variable

File: src/redux/reducers/reducer.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case FILTER_MOVIES_BY_GENRE:
+    case FILTER_MOVIES_BY_QUERY:
+    case SORT_MOVIES:
+      return { ...state, ...action.payload };
+    case DELETE_MOVIE:
+      const movieToDelete = action.payload;
+      const index = MOVIE_LIST.findIndex((movie) => movie === movieToDelete);
```
Comment (alionaSipchuk) [inline src/redux/reducers/reducer.js:29]: why did you use MOVIE_LIST constant, pls use `state.movies`

File: src/redux/reducers/reducer.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+      MOVIE_LIST.splice(index, 1);
```
Comment (alionaSipchuk) [inline src/redux/reducers/reducer.js:30]: ditto

**Review by alionaSipchuk — APPROVED** [review]
pls fix minor issues!)

File: src/redux/selectors.js (hunk 3 above)
Comment (amiteshhh) [reply src/redux/selectors.js:31]: I will move it to store. Actually I am storing filtered/sorted movies in store. However when user clears out the filter or change the genre, I need to filter out on all movies.

**Pull request merged**